sdks/go/pkg/beam/core/runtime/exec/sdf_invokers_test.go

```diff
@@ -382,3 +382,50 @@ func (fn *VetKvSdf) ProcessElement(rt *VetRTracker, i, j int, emit func(*VetRest
 	rest.ProcessElm = true
 	emit(rest)
 }
+
+// VetEmptyInitialSplitSdf runs an SDF in order to test that these methods get called properly,
+// each method will flip the corresponding flag in the passed in VetRestriction,
+// overwrite the restriction's Key and Val with the last seen input elements,
+// and retain the other fields in the VetRestriction.
+type VetEmptyInitialSplitSdf struct {
```

Comment:
Style Nit: This could probably be implemented more succinctly through [type embedding](https://golang.org/doc/effective_go#embedding) (kinda like inheritance, you can use all the methods of VetSdf, just changing SplitRestriction). Not worth blocking the PR on since the code works perfectly fine, just throwing it out there for future reference.

Reply:
That looks so convenient but was unable to get it to work since the ProcessElement method was not being found.

Code I used was:
```
// VetEmptyInitialSplitSdf runs an SDF in order to test that these methods get called properly,
// each method will flip the corresponding flag in the passed in VetRestriction,
// overwrite the restriction's Key and Val with the last seen input elements,
// and retain the other fields in the VetRestriction.
type VetEmptyInitialSplitSdf struct {
  sdf *VetSdf `default:"{}"`
}

// SplitRestriction outputs zero restrictions.
func (fn *VetEmptyInitialSplitSdf) SplitRestriction(i int, rest *VetRestriction) []*VetRestriction {
	return []*VetRestriction{}
}
```

Error I got was:
```
=== RUN   TestSdfNodes

--- FAIL: TestSdfNodes (0.00s)

    sdf_test.go:62: invalid function: 	graph.AsDoFn: for Fn named github.com/apache/beam/sdks/go/pkg/beam/core/runtime/exec.VetEmptyInitialSplitSdf

        failed to find ProcessElement method
```
Will merge as-is. Using the type embedding seems very convenient to have.

Reply:
That sounds about right. The compiler does hoist embeded type methods to the parent type, but the reflect libraries don't automatically hoist methods on embedded types. We'd need to add it to the analysis to get it working. Not hard, just hasn't been required to add.